app/services/PrePopulationService.scala

```diff
@@ -49,4 +60,44 @@ class PrePopulationService @Inject() {
     psrVersionsResponse.reportStatus == ReportStatus.SubmittedAndInProgress
       || psrVersionsResponse.reportStatus == ReportStatus.SubmittedAndSuccessfullyProcessed
   )
+
+  def cleanUpLOPDetails(baseUA: UserAnswers, currentUA: UserAnswers)(srn: Srn): Try[UserAnswers] = {
+
+    val baseUaJson = baseUA.data.decryptedValue
+    val portionStillHeldMap =
+      (baseUaJson \ "assets" \ "landOrProperty" \ "landOrPropertyTransactions" \ "disposedPropertyTransaction" \ "portionStillHeld")
+        .as[Map[String, Map[String, Boolean]]]
+
+    val transformedResult = baseUaJson
+      .transform(landOrProperty.json.pickBranch)
+      .flatMap(_.transform(LandOrPropertyRecordVersionPage(srn).path.prune(_)))
+      .flatMap(_.transform(LandOrPropertyHeldPage(srn).path.prune(_)))
+      .flatMap(_.transform(LandOrPropertyDisposalPage(srn).path.prune(_)))
+      .flatMap(_.transform(disposalPropertyTransaction.prune(_)))
+      .flatMap(_.transform((Paths.heldPropertyTransactions \ "isLandOrPropertyResidential").prune(_)))
+      .flatMap(_.transform((Paths.heldPropertyTransactions \ "landOrPropertyLeased").prune(_)))
```

Comment:
It would be possible to add case class `IsLandOrPropertyResidentialPages` (same as `IsLandOrPropertyResidentialPage` but without the last index part in path). That would avoid  defining `"landOrPropertyLeased"` here. 
What do you think, would it be cleaner or it is an overkill?

Reply:
I removed this 
`      .flatMap(
        _.transform(
          (Paths.heldPropertyTransactions \ "leaseDetails" \ "connectedPartyStatus" \ "isLesseeConnectedParty").prune(_)
        )
      )`
      
This is already covered by 

`.flatMap(_.transform((Paths.heldPropertyTransactions \ "leaseDetails").prune(_)))`

Reply:
That's great. What I meant above is that we can possibly avoid defining e.g. "landOrPropertyLeased" in multiple places: here and in `IsLandOrPropertyResidentialPage`. I.e. if we define it as path in IsLandOrPropertyResidentialPages and use the path here. But it's very minor and might be pros and cons, so resolving.